src/constructs/aws/abstracts/StaticWebsiteAbstract.ts

```diff
@@ -0,0 +1,316 @@
+import type { ConstructCommands } from "@lift/constructs";
+import { AwsConstruct } from "@lift/constructs/abstracts";
+import type { AwsProvider } from "@lift/providers";
+import chalk from "chalk";
+import type { FromSchema } from "json-schema-to-ts";
+import { flatten } from "lodash";
+import { emptyBucket, invalidateCloudFrontCache } from "../../../classes/aws";
+import ServerlessError from "../../../utils/error";
+import type { Progress } from "../../../utils/logger";
+import { getUtils } from "../../../utils/logger";
+import { s3Sync } from "../../../utils/s3-sync";
+
+export const COMMON_STATIC_WEBSITE_DEFINITION = {
+    type: "object",
+    properties: {
+        path: { type: "string" },
+        domain: {
+            anyOf: [
+                { type: "string" },
+                {
+                    type: "array",
+                    items: { type: "string" },
+                },
+            ],
+        },
+        certificate: { type: "string" },
+        security: {
+            type: "object",
+            properties: {
+                allowIframe: { type: "boolean" },
+            },
+            additionalProperties: false,
+        },
+        errorPage: { type: "string" },
+        redirectToMainDomain: { type: "boolean" },
+    },
+    additionalProperties: false,
+    required: ["path"],
+} as const;
+
+type AbstractStaticWebsiteConfiguration = FromSchema<typeof COMMON_STATIC_WEBSITE_DEFINITION>;
+
+export abstract class StaticWebsiteAbstract extends AwsConstruct {
+    public static commands: ConstructCommands = {
+        upload: {
+            usage: "Upload files directly to S3 without going through a CloudFormation deployment.",
+            handler: StaticWebsiteAbstract.prototype.uploadWebsiteCommand,
+        },
+    };
+
+    protected readonly distribution: Distribution;
+    protected readonly domains: string[] | undefined;
+    private readonly bucketNameOutput: CfnOutput;
+    private readonly domainOutput: CfnOutput;
+    private readonly cnameOutput: CfnOutput;
+    private readonly distributionIdOutput: CfnOutput;
+
+    constructor(
+        scope: CdkConstruct,
+        protected readonly id: string,
+        protected readonly configuration: AbstractStaticWebsiteConfiguration,
+        protected readonly provider: AwsProvider
+    ) {
+        super(scope, id);
+
+        if (configuration.domain !== undefined && configuration.certificate === undefined) {
+            throw new ServerlessError(
+                `Invalid configuration for the static website '${id}': if a domain is configured, then a certificate ARN must be configured in the 'certificate' option.\n` +
+                    "See https://github.com/getlift/lift/blob/master/docs/static-website.md#custom-domain",
+                "LIFT_INVALID_CONSTRUCT_CONFIGURATION"
+            );
+        }
+
+        const bucket = new Bucket(this, "Bucket", {
+            // Enable static website hosting
+            websiteIndexDocument: "index.html",
```

Comment:
The commit https://github.com/getlift/lift/commit/e5da8c9daa658dd33395961202fe7900a0bfcaad should be reverted as it is not used anymore for SPA to work properly